lib/surface/formatter/plugin.ex

```diff
@@ -0,0 +1,62 @@
+defmodule Surface.Formatter.Plugin do
```

Comment:
fwiw, it does still work properly; just logs a warning during the compilation of surface_formatter

Reply:
I don't know the way to omit the warning except add the `if` guard like:

```elixir
if Version.match?(System.version(), ">= 1.13.0-rc.0") do
  defmodule Surface.Formatter.Plugin do
    ....
  end
end
```

But the Elixir version 1.12.x or earlier will not see this module. What do you think?

Reply:
Great information and ideas! Leading me to, I think we want this:

```elixir
if Code.ensure_loaded?(Mix.Tasks.Format) do
  defmodule Surface.Formatter.Plugin do
    # ...
  end
end
```

Reply:
After running into CI failures and considering the factors, I concluded we don't need to wrap it in an `if`. Reasons include:

- Specifying `@behaviour` with a module that doesn't define any callback is a compilation warning, not an error.
- Emitting a warning on compilation isn't the end of the world. Lost of Hex packages do it, depending on your Elixir version.
- We can specify in our CI setup which Elixir versions to fail with `warnings-as-errors`.
- `Code.ensure_loaded?` wouldn't work because `Mix.Tasks.Format` has existed for a long time. 😄 I was looking for a solution that didn't require us to specify a specific minimum Elixir version, but didn't find one. I tried `function_exported?(Mix.Tasks.Format, :behaviour_info, 1)`, but that gives `false` even when you can call `Mix.Tasks.Format.behaviour_info(:callbacks)`.